multiqc/modules/stacks/stacks.py

```diff
@@ -0,0 +1,287 @@
+            s_name = self.clean_s_name(os.path.basename(f['root']), run_name)
+            try:
+                self.cov_data.update(self.parse_gstacks(f['f'], s_name, f))
+                num_files += 1
+            except:
+                log.error('Could not parse gstacks.distribs file in {}'.format(f['s_name']))
+
+        # Parse populations data
+        self.distribs_loci = OrderedDict()
+        self.distribs_snps = OrderedDict()
+        for f in self.find_log_files('stacks/populations'):
+            run_name = os.path.dirname(f['root'])
+            s_name = self.clean_s_name(os.path.basename(f['root']), run_name)
+            i,j = self.parse_populations(f['f'], s_name, f)
+            try:
+                self.distribs_loci.update(i); self.distribs_snps.update(j)
+                num_files += 1
+            except:
+                log.error('Could not parse population.log.distribs file in {}'.format(f['s_name']))
+
+        # Parse sumstats file
+        self.sumstats_data = OrderedDict()
+        for f in self.find_log_files('stacks/sumstats'):
+            run_name = os.path.dirname(f['root'])
+            s_name = self.clean_s_name(os.path.basename(f['root']), run_name)
+            try:
+                self.sumstats_data.update(self.parse_sumstats(f['f'], s_name, f))
+                num_files += 1
+            except:
+                log.error('Could not parse populations.sumstats_summary file in {}'.format(f['s_name']))
+
+
+        # Ignore samples
+        self.cov_data = self.ignore_samples(self.cov_data)
+        self.distribs_loci = self.ignore_samples(self.distribs_loci)
+        self.distribs_snps = self.ignore_samples(self.distribs_snps)
+        self.sumstats_data = self.ignore_samples(self.sumstats_data)
+
+        if len(self.cov_data) == 0 and len(self.sumstats_data) == 0 and len(self.distribs_loci) == 0:
+            raise UserWarning
+        log.info("Found {} reports".format(num_files))
+
+        # Write parsed report data to a file
+        self.write_data_file(self.cov_data, 'multiqc_stacks_cov')
+        self.write_data_file(self.sumstats_data, 'multiqc_stacks_sumstats')
+
+        ### Write the sample table
+        config_table = {
+            'id': 'gstacks_table',
+            'namespace': 'stacks'
+        }
+        self.add_section (
+            name = 'Sample statistics',
+            anchor = 'stacks-gstacks',
+            description = 'The sample specific statistics for Stacks',
+            helptext = '''**Note!** The sample names have the following scheme `<run folder name> | <input fastq file prefix>`.
+                        This data is obtained from the gstacks program run after builing sample and catalog loci merge
+                        paired-ends and call variants.
+                        These numbers are obtained from the `gstacks.log.distribs` file''',
+            plot = table.plot(self.cov_data, self.gsheaders, config_table)
```

Comment:
Any reason not to use General Statistics here?

Reply:
I had the vague idea that these samples are ideally in a 1:1 ratio to the raw number of samples. This isn't always the case for Stacks. A common case is to run Stacks several times with varying parameters (where the column values of these samples can vary). Another less common case is that Stacks supports in-read barcodes, so that it can do additional demultiplexing.